lib/compiler.js

```diff
@@ -69,6 +69,8 @@ Compiler.prototype.compile = function () {
     js = `
       function generateLiteralWidget(contents) {
         function LiteralWidget(contents) {
+          this.name = 'LiteralWidget'
+          this.id = contents
```

Comment:
`contents` could be quite large right? How about making id an auto-incrementing int, or at least a hash of contents (calculated of course at compile time not runtime)?